metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1726,8 +1723,26 @@ class MetalsLanguageServer(
           scribe.debug(s"Executing AnalyzeStacktrace ${command}")
         }.asJavaObject
 
-      case ServerCommands.ShowTasty(uri) =>
-        tastyHandler.executeShowTastyCommand(uri).asJavaObject
+      case ServerCommands.ShowTasty(positionParams) =>
+        fileDecoderProvider.showTasty(positionParams).asJavaObject
+
+      case ServerCommands.ShowAvailableClasses(params) =>
```

Comment:
I wonder, is it possible to do this functionality without adding another endpoint? Couldn't we instead do this logic right in the show tasty function? Meaning if the request to show tasty comes in, if there are multiple classes for the given URI, then we send a `metals/quickPick` to get which one they want, and then show it?

Reply:
I think we should go that way, it's always better to add less since that means less work for the clients.

Reply:
It's worth adding, that I want to remove the show tasty command in favor of the `DecodeFile` one and I'm looking for a solution that will suit both cases.

I have some doubts about only one endpoint approach. IMO, from the client's point of view, having two separate commands is more flexible than just one. `decompile` action consists of two separate steps: find a given resource (tasty/class/semanticDB) URI and then translate that resource into human-readable text (the true `decompile` part). 

Especially, I don't know how to handle VirtualDocuments or CustomEditors in vs code with just one command. They both require a provider, which will then provide text content for a given uri. From this point of view, having separate commands seems reasonable.

Reply:
We can do it without the virtual document for now or not differentiate the URLs (show all the classes from one files under one URL for now.), so the quick pick will work the same in all clients.